syz-manager/covfilter.go

```diff
@@ -0,0 +1,213 @@
+	funcs := cfg.CovFilter.Functions
+	rawPCs := cfg.CovFilter.RawPCs
+	if len(files) == 0 && len(funcs) == 0 && len(rawPCs) == 0 {
+		return "", nil
+	}
+	filesRegexp, err := getRegexps(files)
+	if err != nil {
+		return "", err
+	}
+	funcsRegexp, err := getRegexps(funcs)
+	if err != nil {
+		return "", err
+	}
+
+	covFilter := CoverFilter{
+		weightedPCs:    make(map[uint32]uint32),
+		target:         cfg.SysTarget,
+		bitmapFilename: cfg.Workdir + "/" + "syz-cover-bitmap",
+	}
+
+	if len(filesRegexp) > 0 || len(funcsRegexp) > 0 {
+		log.Logf(0, "initialize coverage information...")
+		if err = initCover(cfg.SysTarget, cfg.KernelObj, cfg.KernelSrc, cfg.KernelBuildSrc); err != nil {
+			return "", err
+		}
+		symbols := reportGenerator.GetSymbolsInfo()
+		if err = covFilter.initFilesFuncs(filesRegexp, funcsRegexp, symbols); err != nil {
+			return "", err
+		}
+	}
+
+	if err = covFilter.initWeightedPCs(rawPCs); err != nil {
+		return "", err
+	}
+
+	covFilter.detectRegion()
+	if covFilter.pcSize > 0 {
+		log.Logf(0, "coverage filter from 0x%x to 0x%x, size 0x%x",
+			covFilter.pcStart, covFilter.pcEnd, covFilter.pcSize)
+	} else {
+		return "", fmt.Errorf("coverage filter is enabled but nothing will be filtered")
+	}
+
+	if err = osutil.WriteFile(covFilter.bitmapFilename, covFilter.bitmapBytes()); err != nil {
+		return "", err
+	}
+	return covFilter.bitmapFilename, nil
+}
+
+func getRegexps(regexpStrings []string) ([]*regexp.Regexp, error) {
+	var regexps []*regexp.Regexp
+	for _, rs := range regexpStrings {
+		r, err := regexp.Compile(rs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile regexp: %v", err)
+		}
+		regexps = append(regexps, r)
+	}
+	return regexps, nil
+}
+
+func (covFilter *CoverFilter) initFilesFuncs(filesRegexp, funcsRegexp []*regexp.Regexp, symbols []cover.Symbol) error {
+	fileDedup := make(map[string]bool)
+	used := make(map[*regexp.Regexp][]string)
+	for _, sym := range symbols {
+		matched := false
+		for _, re := range funcsRegexp {
+			if re.MatchString(sym.Name) {
+				matched = true
+				used[re] = append(used[re], sym.Name)
+				break
+			}
+		}
+		for _, re := range filesRegexp {
+			if re.MatchString(sym.File) {
+				matched = true
+				if !fileDedup[sym.File] {
+					fileDedup[sym.File] = true
+					used[re] = append(used[re], sym.File)
+				}
+				break
+			}
+		}
+		if matched {
+			for _, pc := range sym.PCs {
+				covFilter.weightedPCs[uint32(pc)] = 1
+			}
+		}
+	}
+
+	for _, re := range filesRegexp {
+		if _, ok := used[re]; !ok {
+			log.Logf(0, "coverage file filter doesn't match anything: %v", re.String())
+		} else {
+			log.Logf(1, "coverage file filter: %v: %v", re.String(), used[re])
+		}
+	}
+	for _, re := range funcsRegexp {
+		if _, ok := used[re]; !ok {
+			log.Logf(0, "coverage func filter doesn't match anything: %v", re.String())
+		} else {
+			log.Logf(1, "coverage func filter: %v: %v", re.String(), used[re])
+		}
+	}
+	// TODO: do we want to error on this? or logging it enough?
```

Comment:
I had supported trace matched files/functions of per regexp.
In my opinion, either error or log is OK, I prefer to just throw out an error. So, users will realize immediately that the target function isn't enabled, but not after running for a while. 
And write down a correct regexp is the duty of the user, we already print out things matched by per regexp to help users correcting their regexp.
But, it can't stop those people from asking why their syzkaller is broken or complaining it's hard to use. Especially those who just post their questions without reading a word of their log.
So, I think it would be better that you make a decision here, since it may bother you most.